cellbase-lib/src/main/java/org/opencb/cellbase/lib/impl/core/DataReleaseSingleton.java

```diff
@@ -0,0 +1,199 @@
+            for (DataRelease dataRelease : dataReleases) {
+                Map<String, MongoDBCollection> collectionMap = new HashMap<>();
+                for (Map.Entry<String, String> entry : dataRelease.getCollections().entrySet()) {
+                    collectionMap.put(entry.getKey(), releaseMongoDBAdaptor.mongoDataStore.getCollection(entry.getValue()));
+                }
+                cachedData.get(dbname).put(dataRelease.getRelease(), collectionMap);
+            }
+        }
+    }
+
+    public void checkDataRelease(String species, String assembly, String version, int release) throws CellBaseException {
+        checkDataRelease(MongoDBManager.getDatabaseName(species, assembly, version), release);
+    }
+
+    public void checkDataRelease(String dbname, int release) throws CellBaseException {
+        checkDataRelease(dbname, release, null);
+    }
+
+    public void checkDataRelease(String dbname, int release, String data) throws CellBaseException {
+        // Lock and load data if necessary
+        if (!cachedData.containsKey(dbname)) {
+            // If the data release is invalid, throw an exception
+            String msg = UNKOWN_DATABASE_MSG_PREFIX + dbname;
+            throw new CellBaseException(msg);
+        }
+        rwLockMap.get(dbname).writeLock().lock();
+        try {
+            if (!cachedData.get(dbname).containsKey(release)
+                    || (StringUtils.isNotEmpty(data) && !cachedData.get(dbname).get(release).containsKey(data))) {
+                // Load the data releases from the MongoDB collection for that database name
+                loadData(dbname);
+
+                // Check after loading
+                if (!cachedData.get(dbname).containsKey(release)) {
+                    // If the release is invalid, throw an exception
+                    String msg = INVALID_RELEASE_MSG_PREFIX + release + ". The available data releases are: "
+                            + cachedData.get(dbname).keySet();
+                    throw new CellBaseException(msg);
+                }
+                if (StringUtils.isNotEmpty(data) && !cachedData.get(dbname).get(release).containsKey(data)) {
+                    // If the data is invalid, throw an exception
+                    String msg = INVALID_DATA_MSG_PREFIX + " '" + data + "', it's not present in release " + release
+                            + ". The available data are: " + cachedData.get(dbname).get(release).keySet();
+                    throw new CellBaseException(msg);
+                }
+            }
+        } finally {
+            rwLockMap.get(dbname).writeLock().unlock();
+        }
+    }
+
+    // Method to get collection name based on the data and the release
+    public MongoDBCollection getMongoDBCollection(String dbname, String data, int release) throws CellBaseException {
+        checkDataRelease(dbname, release, data);
+        return cachedData.get(dbname).get(release).get(data);
```

Comment:
If "release" or "dbName" do not exist, this will throw an NPE

Reply:
it's checked in the function checkDataRelease